### ISSUE TITLE
[mini.completion] automatically detect file completion

### ISSUE THREAD

**Issue by nat-418** [echasnovski/mini.nvim#308]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)

### Module(s)

mini.completion

### Description

I used to use vim-mucomplete, and I really liked the way it automagically handled fallbacks. Basically, in 99% cases I want to use  LSP completion / omnicompletion or file path completion. File paths almost always begins with a `/`, `.`, or `\` symbol, which is unusual in programming language syntax. 

tl;dr I think mini.completions should fallback to file completion when a `.` or `/` symbol is under the cursor.

**Comment by echasnovski** [comment]
Thanks for suggestion!

> File paths almost always begins with a `/`, `.`, or `\` symbol, which is unusual in programming language syntax.
> 
> tl;dr I think mini.completions should fallback to file completion when a `.` or `/` symbol is under the cursor.

I don't think that this behavior is universally acceptable. For example, many class methods can be thought as "starting with `.`".

------

'mini.completion' is designed around "two-step" completion: first try LSP, then fall back to some other *configurable* source. I'd suggest one of the two solutions:

- Use `<C-x><C-f>` as fallback action. Like this:
```lua
require('mini.completion').setup({
  fallback_action = '<C-x><C-f>`,
})
```
- Write your own action which does what you describe. It can use `vim.fn.getline('.')` and `vim.fn.col('.')` (or similar combination of `vim.api` functions) to compute the branching condition and then `vim.api.nvim_feedkeys(..., 'n', false)` to simulate built-in completion.

Closing, as this is unplanned.

**Comment by nat-418** [comment]
I've never seen a method start with ./

**Comment by echasnovski** [comment]
> begins with a `/`, `.`, or `\` symbol, which is unusual in programming language syntax.

> I've never seen a method start with ./

Oh, so you've then meant `/`, `./`, `\`, or `.\`.

**Comment by nat-418** [comment]
If you think it's too much of a check to add to mini, I can add it myself in config. Just seems a shame to burn my fallback for the occasional path. What I really liked about mucomplete is the fallback chaining. Maybe I will implement that as a custom fallback function. In nearly all cases what I want to complete is LSP if exists, else omnifunc if exists, else word in buffer if exists, else dictionary if exists, else path if exists. So I think a plugin that "does the right thing" would be useful to avoid huge configs for no reason.

**Comment by echasnovski** [comment]
> If you think it's too much of a check to add to mini, I can add it myself in config.

Yes, I feel that this is the most appropriate choice here. Having customizable completion sources outside of current two steps is not planned.
